### DISCUSSION TITLE
webots_ros2_driver

### DISCUSSION THREAD

**Discussion by ntukenmez3** [bitcraze/discussions#894] [Q&A] [answered]
Hi all, we would like to implement our controller developed by our team on CF2 simulator. We got the following error, could you help us to get rid of that problem, thanks.

**~/webots_ros2/webots_ros2_mavic/launch$** ros2 launch ~/webots_ros2/webots_ros2_mavic/launch/robot_launch.py 

[INFO] [launch]: All log files can be found below /home/ntukenmez3/.ros/log/2023-09-05-19-35-21-978685-ae-icps-407120-168970
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Caught exception when trying to load file of format [py]: No module named 'webots_ros2_driver'

**Comment by knmcguire** [comment]
Hi! that seems to be a webots and ROS-specific issue. You should post this on the support channels of ROS.

On the first eye `No module named 'webots_ros2_driver'` can already give a good hint of what is wrong.

**Comment by ntukenmez3** [comment]
I'll ask them, thanks.
________________________________

**Comment by ntukenmez3** [comment]
Hi, tysm firstly, we fixed the previous problem. It works now as in the picture. I have two questions.
1. Why the connection is changing as in the picture (connected \ disconnected)
2. After this point how can we implement our algorithm on simulation environment, I mean how could we send a command and add our controller, thank you for your help.
![image](https://github.com/bitcraze/discussions/assets/143043483/e6cf1976-ffd1-4d9f-b14a-9ecb551b40a9)

**Reply by knmcguire** [reply, marked as answer]
If you look through the error messages on the ROS side of things, you see that the scripts keeps stopping because of those, which is probably causing it to connect or disconnect. 

Do like I do, and google every error you see 😄 Also, this seems to be a python, webots, ROS problem as well, and not related to the current state of the crazyflie simulator.

**Comment by ntukenmez3** [comment]
Thanks, we’ll check again.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ntukenmez3** [comment]
I would like to ask one more thing. Could you please more clarify **the following code** if possible. The problem might be that code. Because we were able to **launch** just one environment. We do **not** have an access to configure `.py` files, thanks.

```
sys.path.append('/home/knmcguire/Development/bitcraze/c/crazyflie-firmware')
import cffirmware
```

**Reply by knmcguire** [reply]
Ah sorry I missed this part. this seems like it should be pointing to the build python bindings of the crazyflie-firmware repo: https://github.com/bitcraze/crazyflie-firmware. But this seems like old code... also there should be a comment that you should change it to the folder of your own repo.

Anyway, Next time that you have a different question, I would advise to make a new discussion because then we won't miss it as easily.

**Comment by knmcguire** [comment]
Just FYI, I've created a pull request of some code I've been working on for the ROS2 node. This will be eventually merged into the webots repo (long overdue I know..) https://github.com/bitcraze/crazyflie-simulation/pull/52

**Reply by ntukenmez3** [reply]
> Just FYI, I've created a pull request of some code I've been working on for the ROS2 node. This will be eventually merged into the webots repo (long overdue I know..) https://github.com/bitcraze/crazyflie-simulation/pull/52

Thank you for letting me know and hope to see more updates soon ☺️
